rust/src/ketama.rs

```diff
@@ -40,13 +42,28 @@ impl KetamaPool {
         };
         self.ranking[ranking_idx % self.ranking.len()].index as usize
     }
+
+    /// Adds the appropriate rankings for the new node
+    pub fn add_node(&mut self, server_name: &str) {
+        create_server_ranking(&[server_name], &mut self.ranking);
+    }
+
+    /// Removes the rankings corresponding to the server name
+    pub fn remove_node(&mut self, server_name: &str) {
+        self.ranking
+            .retain(|ranking| ranking.server_name == server_name)
+    }
+
+    pub fn get_node(&self, key: &str) -> &str {
+        let slot = self.get_slot(key);
+        self.ranking[slot].server_name.as_ref()
+    }
 }
 
 const POINTS_PER_HASH: usize = 4;
 const POINTS_PER_SERVER: usize = 40;
 
-fn create_server_ranking(keys: &[&str]) -> Vec<ServerRank> {
-    let mut ranking = Vec::with_capacity(POINTS_PER_SERVER * POINTS_PER_HASH * keys.len());
+fn create_server_ranking(keys: &[&str], ranking: &mut Vec<ServerRank>) {
```

Comment:
since you are now writing into an out-parameter, renaming this to `update_server_ranking` would make sense